servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCodingHttpRequesterFilter.java

```diff
@@ -0,0 +1,138 @@
+        return new StreamingHttpClientFilter(client) {
+            @Override
+            protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                            final HttpExecutionStrategy strategy,
+                                                            final StreamingHttpRequest request) {
+                return Single.defer(() -> codecTransformBidirectionalIfNeeded(delegate(), strategy, request));
+            }
+        };
+    }
+
+    @Override
+    public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnection connection) {
+        return new StreamingHttpConnectionFilter(connection) {
+            @Override
+            public Single<StreamingHttpResponse> request(final HttpExecutionStrategy strategy,
+                                                         final StreamingHttpRequest request) {
+                return Single.defer(() -> codecTransformBidirectionalIfNeeded(delegate(), strategy, request));
+            }
+        };
+    }
+
+    @Override
+    public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
+        // No influence since we do not block.
+        return strategy;
+    }
+
+    private Single<StreamingHttpResponse> codecTransformBidirectionalIfNeeded(final StreamingHttpRequester delegate,
+                                                                              final HttpExecutionStrategy strategy,
+                                                                              final StreamingHttpRequest request) {
+        final BufferAllocator alloc = delegate.executionContext().bufferAllocator();
+        setAcceptEncoding(request.headers(), acceptedEncodingsHeader);
+        encodePayloadContentIfAvailable(request, alloc);
+
+        return decodePayloadContentIfEncoded(delegate.request(strategy, request), alloc);
+    }
+
+    private Single<StreamingHttpResponse> decodePayloadContentIfEncoded(
+            final Single<StreamingHttpResponse> responseSingle, final BufferAllocator allocator) {
+
+        return responseSingle.map(response -> {
+            ContentCodec coding = identifyContentEncodingOrNullIfIdentity(response.headers(), supportedCodings);
+            if (coding != null) {
+                response.transformPayloadBody(bufferPublisher -> coding.decode(bufferPublisher, allocator));
+            }
+
+            return response;
+        });
+    }
+
+    @Nullable
+    private static CharSequence buildAcceptEncodingsHeader(final List<ContentCodec> codecs) {
+        StringBuilder builder = new StringBuilder();
+        for (ContentCodec enc : codecs) {
+            if (enc == identity()) {
+                continue;
+            }
+
+            if (builder.length() > 0) {
+                builder.append(", ");
+            }
+
+            builder.append(enc.name());
+        }
+
+        return builder.length() > 0 ? builder.toString() : null;
```

Comment:
Because this result will be cached in a variable and will be written multiple time, consider wrapping the retirned `String` with `newAsciiString`. It will let the `HttpObjectEncoder` write it faster.